plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -8804,19 +8804,33 @@ public void testSelectFromPrestoViewReferencingHiveTableWithTimestamps()
     @Test
     public void testTimestampWithTimeZone()
     {
-        String catalog = getSession().getCatalog().orElseThrow();
+        testTimestampWithTimeZone(HiveTimestampPrecision.MILLISECONDS);
+        testTimestampWithTimeZone(HiveTimestampPrecision.MICROSECONDS);
+        testTimestampWithTimeZone(HiveTimestampPrecision.NANOSECONDS);
+    }
+
+    private void testTimestampWithTimeZone(HiveTimestampPrecision timestampPrecision)
+    {
+        Session session = withTimestampPrecision(getSession(), timestampPrecision);
+        String catalog = session.getCatalog().orElseThrow();
+
+        String fractionalPart = switch (timestampPrecision) {
+            case MILLISECONDS -> "123";
+            case MICROSECONDS -> "123456";
+            case NANOSECONDS -> "123456789";
+        };
 
-        assertUpdate("CREATE TABLE test_timestamptz_base (t timestamp) WITH (format = 'PARQUET')");
-        assertUpdate("INSERT INTO test_timestamptz_base (t) VALUES" +
-                     "(timestamp '2022-07-26 12:13')", 1);
+        assertUpdate(session, "CREATE TABLE test_timestamptz_base (t timestamp(" + timestampPrecision.getPrecision() + ")) WITH (format = 'PARQUET')");
+        assertUpdate(session, "INSERT INTO test_timestamptz_base (t) VALUES" +
```

Comment:
let's write all of the fractional part values into the storage `.123` , `.123456`, `.123456789`
to verify that the right value is being read also when using a smaller precision in the session

Reply:
updated the test